Compiling version 2.0.1 from tar ball (v2.0.1.tgz) gives config error (.git directory seems to be missing)

### Platform

Linux

### Platform Version

Ubuntu 24.04.2 LTS

### FreeDV Version

2.0.1

### Steps to Reproduce

  Download tar ball v2.0.1.tgz and untar/unzip 
  $ cd freedv-gui-2.0.1
   $ python3 -m venv rade-venv
   $ . ./rade-venv/bin/activate
   (rade-venv) $ pip3 install torch torchaudio --index-url https://download.pytorch.org/whl/cpu
   (rade-venv) $ pip3 install matplotlib

3. Run build_linux.sh in freedv-gui-2.0.1 directory
       (rade-venv) $ ./build_linux.sh


### Expected Behavior

No errors

### Actual Behavior

fatal: not a git repository (or any of the parent directories): .git   see below

 rade-venv) [boudewijn@boudewijn-desktop:~/DATV/FreeDV/freedv-gui-2.0.1$](mailto:boudewijn@boudewijn-desktop:~/DATV/FreeDV/freedv-gui-2.0.1$) ./build_linux.sh
+ UT_ENABLE=0
+ USE_NATIVE_AUDIO=1
+ BUILD_TYPE=Debug
+ WITH_ASAN=0
+ WITH_RTSAN=0
+ export FREEDVGUIDIR=(rade-venv) [boudewijn@boudewijn-desktop:~/DATV/FreeDV/freedv-gui-2.0.1$](mailto:boudewijn@boudewijn-desktop:~/DATV/FreeDV/freedv-gui-2.0.1$) ./build_linux.sh
+ UT_ENABLE=0
+ USE_NATIVE_AUDIO=1
+ BUILD_TYPE=Debug
+ WITH_ASAN=0
+ WITH_RTSAN=0
+ export FREEDVGUIDIR=/home/boudewijn/DATV/FreeDV/freedv-gui-2.0.1
+ FREEDVGUIDIR=/home/boudewijn/DATV/FreeDV/freedv-gui-2.0.1
+ cd /home/boudewijn/DATV/FreeDV/freedv-gui-2.0.1
+ '[' -d .git ']'
+ mkdir -p build_linux
+ cd build_linux
+ rm -Rf build_codec2-prefix build_rade-prefix CMakeCache.txt CMakeFiles codec2_build codec2_src config.h contrib CPackConfig.cmake CPackSourceConfig.cmake CTestCustom.cmake fargan_config.h freedv-version.txt generated rade_build rade_src soxconfig.h src
+ cmake -DENABLE_RTSAN=0 -DENABLE_ASAN=0 -DCMAKE_BUILD_TYPE=Debug -DUSE_NATIVE_AUDIO=1 -DUNITTEST=0 ..
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Compilation date = XX2025-07-22XX
fatal: not a git repository (or any of the parent directories): .git
CMake Error at cmake/checkgit.cmake:60 (CheckGitWrite):
  CheckGitWrite Function invoked with incorrect arguments for function named:
  CheckGitWrite
Call Stack (most recent call first):
  cmake/checkgit.cmake:84 (CheckGitVersion)
  cmakelists.txt:109 (CheckGitSetup)


-- Git hash:
-- Looking for DBus...
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1")
CMake Warning (dev) at /usr/share/cmake-3.28/Modules/findpackagehandlestandardargs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (DBUS) does
  not match the name of the calling package (DBus).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/finddbus.cmake:59 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmakelists.txt:217 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found DBUS: /usr/include/dbus-1.0;/usr/lib/x86_64-linux-gnu/dbus-1.0/include
-- Found DBus, will attempt to use rtkit for real-time threading
-- Looking for wxWidgets...
-- Found wxWidgets: -L/usr/lib/x86_64-linux-gnu;-pthread;;;-lwx_gtk3u_core-3.2;-lwx_baseu-3.2;-lwx_gtk3u_aui-3.2;-lwx_gtk3u_html-3.2;-lwx_baseu_net-3.2;-lwx_gtk3u_propgrid-3.2 (found version "3.2.4")
-- wxWidgets version: 3.2.4
-- Looking for include file byteswap.h
-- Looking for include file byteswap.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file stddef.h
-- Looking for include file stddef.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file ltdl.h
-- Looking for include file ltdl.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Check size of int
-- Check size of int - done
-- Looking for floor
-- Looking for floor - found
-- Looking for memset
-- Looking for memset - found
-- Looking for pow
-- Looking for pow - found
-- Looking for sqrt
-- Looking for sqrt - found
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for fmemopen
-- Looking for fmemopen - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for vsnprintf
-- Looking for vsnprintf - found
-- Looking for _fseeki64
-- Looking for _fseeki64 - not found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Threads library flags:
-- Finding PulseAudio...
--   PulseAudio library: /usr/lib/x86_64-linux-gnu/libpulse.so
--   PulseAudio headers: /usr/include
-- Looking for hamlib...
--   Hamlib library: /usr/lib/x86_64-linux-gnu/libhamlib.so
--   Hamlib headers: /usr/include
-- Hamlib library found.
-- Using new try_run syntax for CMake 3.28.3
-- Found Hamlib 4.5.5 - NOT enabling Hamlib 4.6 compile workaround.
-- Looking for samplerate...
--   samplerate library: /usr/lib/x86_64-linux-gnu/libsamplerate.so
--   samplerate headers: /usr/include
-- Looking for sndfile...
--   sndfile library: /usr/lib/x86_64-linux-gnu/libsndfile.so
--   sndfile headers: /usr/include
-- Looking for Speex DSP library.
--   Speex DSP headers: /usr/include
--   Speex DSP library: /usr/lib/x86_64-linux-gnu/libspeexdsp.so
System is Linux
-- Looking for dl library.
--   dl library: /usr/lib/x86_64-linux-gnu/libdl.a
-- Build type will be: Debug
-- Configuring incomplete, errors occurred!
/home/boudewijn/DATV/FreeDV/freedv-gui-2.0.1
+ FREEDVGUIDIR=/home/boudewijn/DATV/FreeDV/freedv-gui-2.0.1
+ cd /home/boudewijn/DATV/FreeDV/freedv-gui-2.0.1
+ '[' -d .git ']'
+ mkdir -p build_linux
+ cd build_linux
+ rm -Rf build_codec2-prefix build_rade-prefix CMakeCache.txt CMakeFiles codec2_build codec2_src config.h contrib CPackConfig.cmake CPackSourceConfig.cmake CTestCustom.cmake fargan_config.h freedv-version.txt generated rade_build rade_src soxconfig.h src
+ cmake -DENABLE_RTSAN=0 -DENABLE_ASAN=0 -DCMAKE_BUILD_TYPE=Debug -DUSE_NATIVE_AUDIO=1 -DUNITTEST=0 ..
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Compilation date = XX2025-07-22XX
fatal: not a git repository (or any of the parent directories): .git
CMake Error at cmake/checkgit.cmake:60 (CheckGitWrite):
  CheckGitWrite Function invoked with incorrect arguments for function named:
  CheckGitWrite
Call Stack (most recent call first):
  cmake/checkgit.cmake:84 (CheckGitVersion)
  cmakelists.txt:109 (CheckGitSetup)


-- Git hash:
-- Looking for DBus...
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1")
CMake Warning (dev) at /usr/share/cmake-3.28/Modules/findpackagehandlestandardargs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (DBUS) does
  not match the name of the calling package (DBus).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/finddbus.cmake:59 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmakelists.txt:217 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found DBUS: /usr/include/dbus-1.0;/usr/lib/x86_64-linux-gnu/dbus-1.0/include
-- Found DBus, will attempt to use rtkit for real-time threading
-- Looking for wxWidgets...
-- Found wxWidgets: -L/usr/lib/x86_64-linux-gnu;-pthread;;;-lwx_gtk3u_core-3.2;-lwx_baseu-3.2;-lwx_gtk3u_aui-3.2;-lwx_gtk3u_html-3.2;-lwx_baseu_net-3.2;-lwx_gtk3u_propgrid-3.2 (found version "3.2.4")
-- wxWidgets version: 3.2.4
-- Looking for include file byteswap.h
-- Looking for include file byteswap.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file stddef.h
-- Looking for include file stddef.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file ltdl.h
-- Looking for include file ltdl.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Check size of int
-- Check size of int - done
-- Looking for floor
-- Looking for floor - found
-- Looking for memset
-- Looking for memset - found
-- Looking for pow
-- Looking for pow - found
-- Looking for sqrt
-- Looking for sqrt - found
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for fmemopen
-- Looking for fmemopen - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for vsnprintf
-- Looking for vsnprintf - found
-- Looking for _fseeki64
-- Looking for _fseeki64 - not found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Threads library flags:
-- Finding PulseAudio...
--   PulseAudio library: /usr/lib/x86_64-linux-gnu/libpulse.so
--   PulseAudio headers: /usr/include
-- Looking for hamlib...
--   Hamlib library: /usr/lib/x86_64-linux-gnu/libhamlib.so
--   Hamlib headers: /usr/include
-- Hamlib library found.
-- Using new try_run syntax for CMake 3.28.3
-- Found Hamlib 4.5.5 - NOT enabling Hamlib 4.6 compile workaround.
-- Looking for samplerate...
--   samplerate library: /usr/lib/x86_64-linux-gnu/libsamplerate.so
--   samplerate headers: /usr/include
-- Looking for sndfile...
--   sndfile library: /usr/lib/x86_64-linux-gnu/libsndfile.so
--   sndfile headers: /usr/include
-- Looking for Speex DSP library.
--   Speex DSP headers: /usr/include
--   Speex DSP library: /usr/lib/x86_64-linux-gnu/libspeexdsp.so
System is Linux
-- Looking for dl library.
--   dl library: /usr/lib/x86_64-linux-gnu/libdl.a
-- Build type will be: Debug
-- Configuring incomplete, errors occurred!

### Additional Comments

It seems that the .git directory is missing in the tar ball

Note: cloned source (git clone https://github.com/drowe67/freedv-gui.git)
gives No errors

--- a/USER_MANUAL.md
+++ b/USER_MANUAL.md
@@ -840,6 +840,7 @@ LDPC | Low Density Parity Check Codes - a family of powerful FEC codes
 
 1. Bugfixes:
     * FreeDV Reporter: Use numeric sort for SNR. (PR #979)
+    * Fix compile failure when compiling from tarball.(PR #985)
 2. Enhancements:
     * Add Mic/Speaker volume control to main window. (PR #980)
 

--- a/cmake/CheckGit.cmake
+++ b/cmake/CheckGit.cmake
@@ -30,7 +30,12 @@ function(CheckGitVersion)
         WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
         OUTPUT_VARIABLE GIT_HASH
         OUTPUT_STRIP_TRAILING_WHITESPACE
+        RESULT_VARIABLE GIT_RESULT
         )
+
+    if (GIT_RESULT)
+        set(GIT_HASH "None")
+    endif (GIT_RESULT)
 
     CheckGitRead(GIT_HASH_CACHE)
     if (NOT EXISTS ${post_configure_dir})

